xds/src/test/java/io/grpc/xds/sds/trust/SdsTrustManagerFactoryTest.java

```diff
@@ -0,0 +1,116 @@
+
+  /** client cert. */
+  private static final String CLIENT_PEM_FILE = "src/test/certs/client.pem";
+
+  /** bad server cert. */
+  private static final String BAD_SERVER_PEM_FILE = "src/test/certs/badserver.pem";
+
+  /** bad client cert. */
+  private static final String BAD_CLIENT_PEM_FILE = "src/test/certs/badclient.pem";
+
+  @Test
+  public void factoryConstructorFromFileTest() throws CertificateException, IOException {
+    SdsTrustManagerFactory factory =
+        new SdsTrustManagerFactory(CA_PEM_FILE, /* certContext= */ null);
+    assertThat(factory).isNotNull();
+    TrustManager[] tms = factory.getTrustManagers();
+    assertThat(tms).isNotNull();
+    assertThat(tms).hasLength(1);
+    TrustManager myTm = tms[0];
+    assertThat(myTm).isInstanceOf(SdsX509TrustManager.class);
+    SdsX509TrustManager sdsX509TrustManager = (SdsX509TrustManager) myTm;
+    X509Certificate[] acceptedIssuers = sdsX509TrustManager.getAcceptedIssuers();
+    assertThat(acceptedIssuers).isNotNull();
+    assertThat(acceptedIssuers).hasLength(1);
+    X509Certificate caCert = acceptedIssuers[0];
+    assertThat(caCert).isEqualTo(CertificateUtils.toX509Certificates(CA_PEM_FILE)[0]);
+  }
+
+  @Test
+  public void goodCertCheckServerTrusted() throws CertificateException, IOException {
+    SdsTrustManagerFactory factory =
+        new SdsTrustManagerFactory(CA_PEM_FILE, /* certContext= */ null);
+    SdsX509TrustManager sdsX509TrustManager = (SdsX509TrustManager) factory.getTrustManagers()[0];
+    X509Certificate[] serverChain = CertificateUtils.toX509Certificates(SERVER_1_PEM_FILE);
+    sdsX509TrustManager.checkServerTrusted(serverChain, "RSA");
+  }
+
+  @Test
+  public void goodCertCheckClientTrusted() throws CertificateException, IOException {
+    SdsTrustManagerFactory factory =
+        new SdsTrustManagerFactory(CA_PEM_FILE, /* certContext= */ null);
+    SdsX509TrustManager sdsX509TrustManager = (SdsX509TrustManager) factory.getTrustManagers()[0];
+    X509Certificate[] clientChain = CertificateUtils.toX509Certificates(CLIENT_PEM_FILE);
+    sdsX509TrustManager.checkClientTrusted(clientChain, "RSA");
+  }
+
+  @Test
+  public void badCertCheckServerTrusted_ExpectException() throws CertificateException, IOException {
+    SdsTrustManagerFactory factory =
+        new SdsTrustManagerFactory(CA_PEM_FILE, /* certContext= */ null);
+    SdsX509TrustManager sdsX509TrustManager = (SdsX509TrustManager) factory.getTrustManagers()[0];
+    X509Certificate[] serverChain = CertificateUtils.toX509Certificates(BAD_SERVER_PEM_FILE);
+    try {
+      sdsX509TrustManager.checkServerTrusted(serverChain, "RSA");
+      Assert.fail("no exception thrown");
+    } catch (CertificateException expected) {
+      assertThat(expected)
+          .hasMessageThat()
+          .contains("unable to find valid certification path to requested target");
+    }
+  }
+
+  @Test
+  public void badCertCheckClientTrusted_ExpectException() throws CertificateException, IOException {
```

Comment:
_expectedException. or it should be `checkClientTrusted_badCert()` or `checkClientTrusted_badCert_throwsException()`

Reply:
I am trying to follow a certain naming convention: 

action/condition  _  expected result

Let me know if you think it's not consistent,

Reply:
resolving for now without changing names

Reply:
renamed as per the unit testing naming guideline